lib/elixir/lib/exception.ex

```diff
@@ -1239,28 +1256,57 @@ defmodule TokenMissingError do
         file: file,
         line: line,
         column: column,
+        end_line: end_line,
+        line_offset: line_offset,
         description: description,
+        opening_delimiter: opening_delimiter,
         snippet: snippet
       })
-      when not is_nil(snippet) and not is_nil(column) do
+      when not is_nil(snippet) and not is_nil(column) and not is_nil(end_line) do
+    lines = snippet |> String.trim_trailing("\n") |> String.split("\n")
+    trimmed_lines = end_line - Enum.count(lines)
+    actual_end_line = end_line - trimmed_lines
+
+    end_column =
+      lines
+      |> Enum.fetch!(actual_end_line - 1)
+      |> String.length()
+      |> Kernel.+(1)
+
+    start_pos = {line, column}
+    end_pos = {actual_end_line, end_column}
+    expected_delimiter = :elixir_tokenizer.terminator(opening_delimiter)
```

Comment:
Let's include it in the exception so we don't need to fetch it back?

Reply:
Actually, nevermind, we do not always have an expected delimiter. It would have to be optional, like endline, but having one optional field is already enough.